### ISSUE TITLE
After release of version 1.2.1, GenericRelation items are no longer safe delete cascaded

### ISSUE THREAD

**Issue by lifenautjoe** [makinacorpus/django-safedelete#207]
On version 1.1.2, items who had a GenericRelation, when they were soft deleted, the item at the other end, also got soft deleted.

After version 1.2.1, the items at the other end of the relationship are no longer soft deleted.

Example between `Notification` and `PostCommentReplyNotification`

````python
class Notification(BaseUserContentModel):
    id = models.AutoField(primary_key=True)

    # Generic relation types
    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
    object_id = models.PositiveIntegerField()
    content_object = GenericForeignKey()
````

````python
class PostCommentReplyNotification(BaseUserContentModel):
    id = models.AutoField(primary_key=True)
    notification = GenericRelation(
        Notification, related_name="post_comment_reply_notifications"
    )
````

When an instance of `PostCommentReplyNotification` which has a "parent" `notification` is deleted. The parent is not deleted as well.

**Comment by Gagaro** [comment]
Thanks for the report :+1: .

I was able to reproduce the issue and fix it.

**Comment by Gagaro** [comment]
Released in 1.2.2

**Comment by lifenautjoe** [comment]
Thanks Gagaro! Great work!
